pkg/apis/serving/v1/revision_helpers.go

```diff
@@ -87,6 +87,39 @@ func (rs *RevisionSpec) GetContainer() *corev1.Container {
 	return &corev1.Container{}
 }
 
+// RoutingState represents states of a revision with regards to serving a route.
+type RoutingState string
+
+const (
+	// RoutingStatePending is a state after a revision is created, but before
+	// its routing state has been determined. It is treated like active for the purposes
+	// of revision garbage collection.
+	RoutingStatePending RoutingState = "pending"
+
+	// RoutingStateActive is a state for a revision which are actively referenced by a Route.
+	RoutingStateActive RoutingState = "active"
+
+	// RoutingStateReserve is a state for a revision which is no longer referenced by a Route,
+	// and is scaled down, but may be rapidly pinned to a route to be made active again.
+	RoutingStateReserve RoutingState = "reserve"
+)
+
+// SetRoutingState sets the routingState label on this Revision and updates the
```

Comment:
s/routingState/RoutingState It isn't clear to me how to be consistent with capitalization. It appears there is a type RoutingState, but also in other cases we do use lower case.

Reply:
The type is "RoutingState", but the annotation is "knative.dev/routingState" - here I was referring to the latter. Capitalization is a funny thing in golang because it has semantic meaning for scope...